contracts/src/token/psp22/extensions/capped.rs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2012-2022 Supercolony
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the"Software"),
+// to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+//
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+pub use crate::{
+    psp22::*,
+    traits::psp22::extensions::capped::*,
+};
+pub use derive::PSP22CappedStorage;
+use ink_prelude::string::String;
+use openbrush::{
+    declare_storage_trait,
+    traits::{
+        AccountId,
+        Balance,
+        InkStorage,
+    },
+};
+
+pub const STORAGE_KEY: [u8; 32] = ink_lang::blake2x256!("openbrush::PSP22CappedData");
+
+#[derive(Default, Debug)]
+#[openbrush::storage(STORAGE_KEY)]
+pub struct PSP22CappedData {
+    pub cap: Balance,
+    pub _reserved: Option<()>,
+}
+
+declare_storage_trait!(PSP22CappedStorage);
+
+impl<T: PSP22CappedStorage<Data = PSP22CappedData> + PSP22 + PSP22Internal + InkStorage> PSP22Capped for T {
+    default fn cap(&self) -> Balance {
+        self.get().cap
+    }
+
+    default fn mint(&mut self, account: AccountId, amount: Balance) -> Result<(), PSP22Error> {
```

Comment:
You don't need to add a new `mint` method=) You need to override the old ones. But it also is a bad idea. Better to have an internal method that does the check and if someone wants a `capped` restriction he should call that method. That method can get `from` and `value` and do the check